src/corredor/helpers/shared.ts

```diff
@@ -32,12 +33,22 @@ export function extractID (value, key) {
   return value
 }
 
-export function isFresh (ID) {
+export function isFresh (ID: string): boolean {
   return !ID || ID === '0'
 }
 
-export function genericPermissionUpdater (API, rules) {
-  const g = rules.reduce((acc, p) => {
+interface Rule {
+  role: {
+    roleID: string;
+  };
+}
+
+interface Permissions {
+  [key: string]: Rule[];
+}
+
+export function genericPermissionUpdater (API: System|Compose|Messaging, rules: Rule[]) {
```

Comment:
how about if you flip this around: remove import of these 3 and add an interface.

simething like:
```
interface PermissionUpdater {
  permissionUpdate({ rule: string; rules: .... });
}
```